gtda/plotting/plot_point_clouds.py

```diff
@@ -0,0 +1,244 @@
+        dictionaries containing keyword arguments as would be fed to the
+        :meth:`update_traces` and :meth:`update_layout` methods of
+        :class:`plotly.graph_objects.Figure`.
+
+    marker_size : float or None, optional, default: 5
+        Sets the size of the markers in the plot. Must be a positive number.
+
+    opacity : float or None, optional, default: 0.8
+        Sets the opacity of the markers in the plot. Must be a number between
+        0 and 1.
+
+    to_scale : bool or None, optional, default: False
+        Whether or not to use the same scale across all axes of the plot.
+
+    show_plot: bool or None, optional, default: True
+        Whether or not to display the plot.
+
+    Returns
+    -------
+    fig : :class:`plotly.graph_objects.Figure` object
+        Figure representing a point cloud in 2D or 3D.
+
+    """
+
+    # If no labels provided, just enumerate data points, and record
+    # if there were user-provided labels to use in `names`.
+    labels_were_provided = labels is not None
+    if not labels_were_provided:
+        labels = np.arange(point_cloud.shape[0])
+
+    validate_params({"labels": labels},
+                    {"labels": {"type": (np.ndarray,),
+                                "of": {"type": (np.number,)}}})
+    validate_params({"names": names},
+                    {"names": {"type": (dict, type(None))}})
+    validate_params({"dimension": dimension},
+                    {"dimension": {"type": (int, type(None)),
+                                   "in": [2, 3]}})
+    validate_params({"marker_size": marker_size},
+                    {"marker_size": {"type": (float, int),
+                                     "in": Int(0, np.inf, closed="neither")}})
+    validate_params({"opacity": opacity},
+                    {"opacity": {"type": (float, int),
+                                 "in": Int(0, 1, closed="right")}})
+    validate_params({"to_scale": to_scale},
+                    {"to_scale": {"type": (bool,)}})
+    validate_params({"display_plot": display_plot},
+                    {"display_plot": {"type": (bool,)}})
+
+    if dimension is None:
+        dimension = np.min((3, point_cloud.shape[1]))
+
+    if names is not None:
+        if not labels_were_provided:
+            raise ValueError("No lables were provided.")
+        all_labels_have_names = (
+            np.array(
+                [label in names.keys() for label in np.unique(labels)]
+            ).all()
+            )
+        if not all_labels_have_names:
+            raise ValueError(
+                "One or more labels are lacking a corresponding name."
+                )
+        all_names_are_strings = (
+            np.array(
+                [type(value) == str for value in names.values()]
+                ).all()
+        )
+        if all_names_are_strings:
+            raise TypeError(
+                "All values of `names` should be strings."
```

Comment:
You can have a better error message. All types should be string or all values of names should be of type string. It is a TypeError, i.e., you should speak about type in your message.